### PR TITLE
[m2e] more safely refresh entire targetDir

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

File: bndtools.m2e/src/bndtools/m2e/BndConfigurator.java [hunk 1]
```diff
@@ -145,20 +150,15 @@ public AbstractBuildParticipant getBuildParticipant(final IMavenProjectFacade pr
...
-				// plugin by reading the classesDir property.
-				final File classesDir = maven.getMojoParameterValue(projectFacade.getMavenProject(), execution,
-					"classesDir", File.class, monitor);
+
+				Set<IPath> deltaFilePaths = collectResourceDeltas(getDelta(project)).map(IResourceDelta::getResource)
+					.filter(IFile.class::isInstance)
+					.map(IResource::getFullPath)
+					.collect(Collectors.toSet());
+
+				final String targetDirectory = projectFacade.getMavenProject()
```
Comment (bjhargrave) [inline bndtools.m2e/src/bndtools/m2e/BndConfigurator.java:159]: Doesn't this lose checking executions for different target directories as the removed comment says?

Comment (gamerson) [reply bndtools.m2e/src/bndtools/m2e/BndConfigurator.java:159]: Yes. I discussed with Ray and he agreed this more "course grained" approach would be ok provided we didn't his the "infinite build" loop problem, which I have tried to address with the new resource delta check.

Comment (rotty3000) [reply bndtools.m2e/src/bndtools/m2e/BndConfigurator.java:159]: @bjhargrave you might be right, but @gamerson and I discussed this and this change solves some problems that @kriegfrj reported and it's on me to test if this breaks the use case I documented in the comment.

Comment (bjhargrave) [reply bndtools.m2e/src/bndtools/m2e/BndConfigurator.java:159]: OK, as long as you got this! :-)

**Review by rotty3000 — APPROVED** [review]
LGTM (tested with osgi-test, aries-cdi, aries-jaxrs-whiteboard)

**Pull request merged**